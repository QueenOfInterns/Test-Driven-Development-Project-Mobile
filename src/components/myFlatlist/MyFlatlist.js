import React, { Component } from 'react';
import { ScrollView, View, Text, StyleSheet} from 'react-native';

class MyFlatlist extends Component {
    constructor(props) {
        super(props);
    
        if (props.data) {
          this.state = {
            users: props.data.users,
            error: props.error,
          };
        } else {
          this.state = {
            users: [],
            error: true,
          };
        }
      }
      render() {
        if(this.state.error == false) {
            return (
                <View>
                    <ScrollView>
                    {
                        this.state.users.map((item) => (
                        <View key = {item.id} style = {styles.item}>
                            <Text>{item.name}</Text>
                            <Text>{item.phoneNumber}</Text>
                        </View>
                        ))
                    }
                    </ScrollView>
                </View>
            );
        }
        else {
            return (
                <View>
                    <Text>ERROR</Text>
                </View>
            );
        }
    }
}


export default MyFlatlist;

const styles = StyleSheet.create ({
    item: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        padding: 45,
        margin: 2,
        borderColor: '#2a4944',
        borderWidth: 1,
        backgroundColor: '#d2f7f1'
    }
})