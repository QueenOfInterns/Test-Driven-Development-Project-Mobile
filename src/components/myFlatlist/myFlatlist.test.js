import MyFlatlist from "./myFlatlist";
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';

describe('MyFlatlist', () => {
    describe('Custom Flatlist', () => {
      it('can render with the given data', () => {
        const props = {
          users: [
            {name: 'Spongebob', phoneNumber: '479-387-5534'},
            {name: 'Patrick', phoneNumber: '479-256-5344'},
            {name: 'Sandy', phoneNumber: '367-899-7070'},
            {name: 'Squidward', phoneNumber: '367-345-9999'},
            {name: 'Mr. Krabs', phoneNumber: '504-214-3322'},
            {name: 'Gary', phoneNumber: '601-405-9977'},
          ],
        };
        const renderer = new ShallowRenderer();
        renderer.render(<MyFlatlist {...props} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
  
      it('does not crash with no data', () => {
        const props = {
          data: [],
        };
  
        const renderer = new ShallowRenderer();
        renderer.render(<MyFlatlist {...props} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
      it('displays error when there is an error', () => {
        const props = {
          error: true,
        };
        const renderer = new ShallowRenderer();
        renderer.render(<MyFlatlist {...props} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
    });
  });