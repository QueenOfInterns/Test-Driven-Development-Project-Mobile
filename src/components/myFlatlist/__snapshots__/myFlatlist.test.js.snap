// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFlatlist Custom Flatlist can render with the given data 1`] = `
<View>
  <Text>
    ERROR
  </Text>
</View>
`;

exports[`MyFlatlist Custom Flatlist displays error when there is an error 1`] = `
<View>
  <ScrollViewMock />
</View>
`;

exports[`MyFlatlist Custom Flatlist does not crash with no data 1`] = `
<View>
  <Text>
    ERROR
  </Text>
</View>
`;
